tests/std/tests/VSO_0000000_allocator_propagation/test.cpp

```diff
@@ -34,6 +34,188 @@ _CONSTEXPR20 void assert_is_permutation(const Container& cont, initializer_list<
     assert(is_permutation(cont.begin(), cont.end(), il.begin(), il.end()));
 }
 
+class lifetime_checker {
+private:
+    static unordered_set<const void*> record;
+
+public:
+    _CONSTEXPR20 lifetime_checker() noexcept /* terminates */ {
+        if (!_Is_constant_evaluated()) {
+            assert(record.insert(this).second);
+        }
+    }
+    _CONSTEXPR20 ~lifetime_checker() {
+        if (!_Is_constant_evaluated()) {
+            assert(record.erase(this) == 1);
+        }
+    }
+
+    _CONSTEXPR20 void assert_living() const noexcept {
+        if (!_Is_constant_evaluated()) {
+            assert(record.find(this) != record.end());
+        }
+    }
+
+    static bool none_living() noexcept {
+        return record.empty();
+    }

```

Comment:
Instead of using `assert_none_living`, I'm keeping the `assert(lifetime_checker::none_living)` pattern in the main function to make it easier to locate bugs between test functions.